src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
@theofidry listing all exceptions explicitly is unmaintainable, if you account for all your dependencies. Any method in PHP can throw an exception when something goes wrong.
they deserve to be documented this way only when there is something useful about it (specific exception classes being used for specific errors which are likely to be interesting to be catched by the caller). ``@throws Exception`` in the DecoderInterface does not bring anything useful IMO.